[DOC] Docs improvements for configuring SCRAM-SHA-512 authentication in Kafka Connect (#3560)
strimzi/strimzi-kafka-operator: documentation/modules/con-kafka-connect-authentication.adoc

[id='con-kafka-connect-authentication{context}']
= Authentication support in Kafka Connect

You can configure authentication for Kafka Connect in the `authentication` property in `KafkaConnect.spec` and `KafkaConnectS2I.spec`. 
The `authentication` property specifies which of the supported authentication mechanisms to use, and additional configuration depending on the mechanism.

The following types of authentication are supported for Kafka Connect:

* TLS client authentication
* SASL-based authentication using the SCRAM-SHA-512 mechanism
* SASL-based authentication using the PLAIN mechanism
* xref:assembly-oauth-authentication_str[OAuth 2.0 token based authentication]

The following sections explain how each authentication mechanism works. 

> == TLS Client authentication

Reviewer: I'm not sure we need this section. I think we could just cross-ref to https://strimzi.io/docs/operators/master/using.html#type-KafkaClientAuthenticationTls-reference as we're moving descriptions of properties into the appendix.

> To use TLS client authentication, set the `authentication.type` property to `tls`.

Reviewer 2: ```suggestion
To use TLS client authentication, set the `spec.authentication.type` property to `tls`.
```

TLS client authentication uses a TLS certificate to authenticate.
The certificate is specified in the `certificateAndKey` property and is always loaded from a Kubernetes Secret.

> In the Secret, the certificate must be stored in X509 format under two different keys: public and private.

Reviewer 2: Use of "keys" here is ambiguous. And TBH I'm not sure I follow what this even means. A certificate is a data structure which includes a public key and some other information. A certificate does not include a private key. So the sentence doesn't make sense because it says the certificate needs to be stored under a private "key" (aka field or property), but there's nothing private about the cert.

```suggestion
In the Secret, the certificate must be stored in X509 format under the field named by the `certificate` and the private key must be stored under the field named by the `key` property, as shown below.
```

  # ...
----

To use this method, see xref:proc-configuring-kafka-connect-authentication-tls-{context}[].

> == SASL-based SCRAM-SHA-512 authentication

Reviewer: Same with TLS. We could just refer to https://strimzi.io/docs/operators/master/using.html#type-KafkaClientAuthenticationScramSha512-reference. If so, I think that section would need a bit more detail from here

> To configure Kafka Connect to use SASL-based SCRAM-SHA-512 authentication, set the `authentication.type` property to `scram-sha-512`.

Reviewer 2: ```suggestion
To configure Kafka Connect to use SASL-based SCRAM-SHA-512 authentication, set the `spec.authentication.type` property to `scram-sha-512`.
```

> This authentication mechanism requires a client username and password, which are contained in a Secret.

Reviewer 3: It is only the password what is in secret. The username is directly in the CR (see the example below.).

> This authentication mechanism requires a client username and password. The password is contained in a Secret.

Reviewer 2: ```suggestion
This authentication mechanism requires a client username and password. 
The password is contained in a Secret. The name of the Secret is given as the value of the `secretName` property and the field within the Secret is given by the `password` property.
```

  # ...
  authentication:
    type: scram-sha-512
    username: my-connect-username <1>
    passwordSecret:
      secretName: my-connect-secret-name <2>
      password: my-connect-password-field <3>
  # ...
----

<1> `username`: The username.
<2> `secretName`: The name of the Secret that contains the username and password.

> <3> `password`: The name of the field in the Secret under which the password is stored.

Reviewer 3: Should we maybe add also an example of secret YAML with the password to make a better connection what should be set there?

Author: Good suggestion. I've added some example YAML (see below). The password is opaque.

```
kind: Secret
apiVersion: v1
metadata:
  name: my-connect-secret-name
  namespace: default
  selfLink: /api/v1/namespaces/default/secrets/my-connect-secret-name
  uid: 77261fa0-00e6-48d4-b4cd-52b20159d83d
  resourceVersion: '32419'
  creationTimestamp: '2020-08-28T14:40:46Z'
data:
  my-connect-password-field: QTxifR4uQ281cnQxMjU=
type: Opaque
```

Author: I added the following example. Please can you check it.

**Example Secret YAML for SCRAM-SHA-512 client authentication for Kafka Connect**
```
apiVersion: v1
kind: Secret
metadata:
  name: my-connect-secret-name
type: Opaque
data:
  my-connect-password-field: LFTIyFRFlMmU2N2Tm
```

[WARNING]
====
Do NOT specify the actual password in the `passwordSecret.password` property.
====

To use this method, see xref:proc-configuring-kafka-connect-authentication-scram-sha-512-{context}[].

> == SASL-based PLAIN authentication

Reviewer: Same here. I think we can just link to the appendix description https://strimzi.io/docs/operators/master/using.html#type-KafkaClientAuthenticationPlain-reference. Modify with any changes from this update
